net-fs/openafs/openafs-1.8.13.ebuild

```diff
@@ -0,0 +1,342 @@
+
+	# pam_afs and pam_afs.krb have been installed in irregular locations, fix
+	if use kauth; then
+		dopammod "${ED}"/usr/$(get_libdir)/pam_afs*
+	fi
+	rm -f "${ED}"/usr/$(get_libdir)/pam_afs* || die
+
+	# remove kdump stuff provided by kexec-tools #222455
+	rm -rf "${ED}"/usr/sbin/kdump* || die
+
+	# avoid collision with mit_krb5's version of kpasswd
+	if use kauth; then
+		mv "${ED}"/usr/bin/kpasswd{,_afs} || die
+		mv "${ED}"/usr/share/man/man1/kpasswd{,_afs}.1 || die
+	fi
+
+	# avoid collision with heimdal's pagsh
+	if has_version app-crypt/heimdal; then
+		mv "${ED}"/usr/bin/pagsh{,_afs} || die
+		mv "${ED}"/usr/share/man/man1/pagsh{,_afs}.1 || die
+	fi
+
+	# move lwp stuff around #200674 #330061
+	mv "${ED}"/usr/include/{lwp,lock,timer}.h "${ED}"/usr/include/afs/ || die
+	mv "${ED}"/usr/$(get_libdir)/liblwp* "${ED}"/usr/$(get_libdir)/afs/ || die
+	# update paths to the relocated lwp headers
+	sed -ri \
+		-e '/^#include <(lwp|lock|timer).h>/s:<([^>]*)>:<afs/\1>:' \
+		"${ED}"/usr/include/*.h \
+		"${ED}"/usr/include/*/*.h \
+		|| die
+
+	# minimal documentation
+	use kauth && doman src/pam/pam_afs.5
+	DOCS=( "${FILESDIR}/README.Gentoo" src/afsd/CellServDB NEWS README )
+
+	# documentation package
+	rm -rf doc/txt/winnotes || die # unneeded docs
+	if use doc; then
+		DOCS+=( doc/{pdf,protocol,txt} CODING CONTRIBUTING )
+		newdoc doc/xml/AdminGuide/auagd000.pdf AdminGuide.pdf
+		newdoc doc/xml/AdminRef/auarf000.pdf AdminRef.pdf
+		newdoc doc/xml/QuickStartUnix/auqbg000.pdf QuickStartUnix.pdf
+		newdoc doc/xml/UserGuide/auusg000.pdf UserGuide.pdf
+	fi
+	use apidoc && DOCS+=( doc/doxygen/output/html )
+	einstalldocs
+
+	# Gentoo related scripts
+	newinitd "${OPENRCDIR}"/openafs-client.initd openafs-client
+	newconfd "${OPENRCDIR}"/openafs-client.confd openafs-client
+	newinitd "${OPENRCDIR}"/openafs-server.initd openafs-server
+	newconfd "${OPENRCDIR}"/openafs-server.confd openafs-server
+	dotmpfiles "${SYSTEMDDIR}"/tmpfiles.d/openafs-client.conf
+	systemd_dounit "${SYSTEMDDIR}"/openafs-client.service
+	systemd_dounit "${SYSTEMDDIR}"/openafs-server.service
+	systemd_install_serviced "${SYSTEMDDIR}"/openafs-client.service.conf
+	systemd_install_serviced "${SYSTEMDDIR}"/openafs-server.service.conf
+
+	# used directories: client
+	keepdir /etc/openafs
+
+	# used directories: server
+	keepdir /etc/openafs/server
+	diropts -m0700
+	keepdir /var/lib/openafs
+	keepdir /var/lib/openafs/db
+	diropts -m0755
+	keepdir /var/lib/openafs/logs
+
+	# link logfiles to /var/log
+	dosym ../lib/openafs/logs /var/log/openafs
+}
+
+pkg_preinst() {
+	## Somewhat intelligently install default configuration files
+	## (when they are not present)
+	local x
+	for x in cacheinfo CellServDB ThisCell ; do
+		if [[ -e "${EROOT}"/etc/openafs/${x} ]] ; then
+			cp "${EROOT}"/etc/openafs/${x} "${ED}"/etc/openafs/
+		fi
+	done
+}
+
+pkg_postinst() {
+	if use modules; then
+		# Update linker.hints file
+		use kernel_linux && linux-mod-r1_pkg_postinst
```

Comment:
Checking linux-mod-r1.eclass, we don't have to do the `if use modules` bit, it does that for us (it gates execution in each phase with the relevant USE check if MODULES_OPTIONAL_IUSE is set).